### PR TITLE
[IMP] core: use tools.config['test_enable'] and current_test

### PR DESCRIPTION
**Pull request by kmagusiak**
A flag on the configuration that indicates whether Odoo was started in a test mode.
However, sometimes it is enough to check that we are in the test mode.

task-4270485
odoo/enterprise#73654
odoo/design-themes#1009

Follow-up of #186884

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Julien00859** [comment]
A property on the config object looks good to me. Not too sure about the name, but I have nothing better that comes to mind.

**Comment by kmagusiak** [comment]
@Xavier-Do rebased and removed the config entry that is no longer needed since #190080 is merged.
Could you review/r+?

**Comment by Xavier-Do** [comment]
Can you rephrase de pr message to indicate the updated changes? :)

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -6249,4 +6249,4 @@ def _can_commit():
 
         :returns: True if commit is acceptable, False otherwise.
         """
-        return not tools.config['test_enable'] and not modules.module.current_test
+        return not (tools.config['test_enable'] or modules.module.current_test)
```
[the remark this answers is not in the document]
Comment (kmagusiak) [reply addons/account/models/account_move.py]: I removed this and reviewed the other ones.

File: addons/base_geolocalize/models/res_partner.py [hunk 2]
```diff
@@ -32,9 +32,12 @@ def _geo_localize(self, street='', zip='', city='', state='', country=''):
...
         # We need country names in English below
-        if not self._context.get('force_geo_localize') \
-                and (self._context.get('import_file') \
-                     or any(config[key] for key in ['test_enable', 'test_file', 'init', 'update'])):
+        if not self._context.get('force_geo_localize') and (
+            self._context.get('import_file')
+            or modules.module.current_test
+            or config['test_enable']
+            or any(config[key] for key in ['init', 'update'])
+        ):
```
Comment (Xavier-Do) [inline addons/base_geolocalize/models/res_partner.py:40]:
```suggestion
        if not self._context.get('force_geo_localize') and (
            self._context.get('import_file')
            or modules.module.current_test
            or not self.env.registry.ready
        ):
```

maybe something like that  (not sure, to check and test)

**Review by Xavier-Do — CHANGES_REQUESTED** [review]
Except this last comment LGTM

**Comment by Xavier-Do** [comment]
@robodoo r+

**Comment by kmagusiak** [comment]
Security: all gettting the testing flag from the current thread (as before)
- addons/sms/models/sms_sms.py:113
- addons/sms/models/sms_sms.py:163
- addons/stock/models/res_company.py:196

**Comment by rco-odoo** [comment]
@robodoo override=ci/security

`getattr(threading.current_thread(), 'testing', False)` is used on a non-forgeable object (it does not depend on local variables) and a static attribute name.  Hence it is safe.
